[leroychiu20] iP

leroychiu20 opened the pull request:
# **Alex**
> "A place for everything, and everything in its place." - Benjamin Franklin [source](https://www.goodreads.com/quotes/175428-a-place-for-everything-everything-in-its-place)

Alex is a chatbot that will surely help you manage your tasks better.

### Why Alex?
Alex is:
- *Intuitive*
- *Free to use*
- *~~Fast~~ SUPER FAST to use 💨💨* 

### What are some things you can do with Alex?

1. Add / delete tasks
2. Mark / unmark tasks
3. Show list of tasks
4. Find tasks with a common keyword
5. ... and many more!

### Features

- [ ] Manage tasks
- [ ] Manage deadlines
- [ ] Manage events

## Main method:
```ruby
public static void main(String[] args) {
    Alex alex = new Alex("./data/alex.txt");
    alex.ui.welcomeMessage();
    alex.scan();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

leroychiu20 committed:
rename greet and exit

leroychiu20 committed:
echoing

leroychiu20 committed:
add list

leroychiu20 committed:
marking and unmarking

leroychiu20 committed:
todo, event, deadline

leroychiu20 committed:
Automated testing

leroychiu20 committed:
error handling

leroychiu20 committed:
delete function

leroychiu20 committed:
Add save features to the chatbot

leroychiu20 committed:
Merge branch 'branch-Level-7'

leroychiu20 committed:
Understand dates as input

leroychiu20 committed:
Catch DateTimeParseException to deadline and event tasks

leroychiu20 committed:
Add response to date commands

leroychiu20 committed:
Merge branch 'branch-Level-8'

leroychiu20 committed:
Add more classes for OOP

leroychiu20 committed:
Merge branch 'branch-A-MoreOOP'

leroychiu20 committed:
Add packages to code

leroychiu20 committed:
Merge branch 'add-gradle-support'

leroychiu20 committed:
Add JUnit tests

leroychiu20 committed:
Create JAR file

leroychiu20 committed:
Add JavaDoc to the relevant classes and methods

leroychiu20 committed:
Follow coding standards

leroychiu20 committed:
Revert "Follow coding standards"

This reverts commit 2c7fe1db48eafc3d74e8dc329bbaceae82120d47.

leroychiu20 committed:
Follow coding standards

leroychiu20 committed:
Add find command

leroychiu20 committed:
Merge branch 'branch-A-JavaDoc'

leroychiu20 committed:
Merge branch 'branch-A-CodingStandard'

leroychiu20 committed:
Merge branch 'branch-Level-9'

dominopizzaaaa commented on src/main/java/AlexExceptions.java:
```diff
@@ -0,0 +1,8 @@
+    public AlexExceptions(String message) {
+        super(message);
+    }
+    public void InvalidInputException(String message) {
```
I think this is supposed to be camelCase since its a method name 🤔

dominopizzaaaa commented on src/main/java/AlexExceptions.java:
```diff
@@ -0,0 +1,8 @@
+public class AlexExceptions extends Exception {
```
The class AlexExceptions extends Exception, but the name AlexExceptions is too generic and does not clearly describe what kind of exceptions it handles. 😠

dominopizzaaaa commented on src/main/java/alex/task/Task.java:
```diff
@@ -0,0 +1,55 @@
+ * a piece of work represented by a string e.g., read book
+ */
+public class Task {
+    public String description;
```
maybe make them to private? 🤔

dominopizzaaaa commented on src/main/java/alex/task/Task.java:
```diff
@@ -0,0 +1,55 @@
+ */
+public class Task {
+    public String description;
+    public boolean isDone;
```
Good job with the naming here! 👍

dominopizzaaaa commented on src/main/java/alex/storage/Storage.java:
```diff
@@ -0,0 +1,109 @@
+public class Storage {
+    public ArrayList<Task> tasks;
+
+    public Storage(String directory, String file) {
```
use filePath instead of file to help others understand your code 🦖

dominopizzaaaa reviewed (commented):
Good job overall, just some minor improvements you can make!

ignatiusoey09 commented on src/main/java/alex/task/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents an action to be done by a certain date. A Deadline object corresponds to
```
Great using header comments, really makes things clearer :)

ignatiusoey09 commented on src/main/java/alex/task/Task.java:
```diff
@@ -0,0 +1,55 @@
+    public String toBeSavedAsData() {
+        return toString();
+    }
+    /**
```
Impressed that you left JavaDocs on everything!

ignatiusoey09 commented on src/test/java/alex/storage/StorageTest.java:
```diff
@@ -0,0 +1,38 @@
+
+public class StorageTest {
+    @Test
+    public void saveTasksToFile_validDate_success() {
```
Great test case naming!

ignatiusoey09 commented on src/main/java/alex/Alex.java:
```diff
@@ -0,0 +1,90 @@
+        Scanner scanner = new Scanner(System.in);
+        String userInput = scanner.nextLine();
+
+        if (userInput.equalsIgnoreCase("bye")) {
```
Perhaps could be written as a switch case statement? Might be easier to read.

ignatiusoey09 commented on src/main/java/alex/Alex.java:
```diff
@@ -0,0 +1,90 @@
+
+
+    public Alex(String filePath) {
+        this.filePath = filePath;
```
Could look at java Paths class instead of using Strings when dealing with file paths to avoid environment related errors.

ignatiusoey09 commented on src/main/java/alex/storage/Storage.java:
```diff
@@ -0,0 +1,109 @@
+                } else {
+                    String details = desc.substring(6);
+                    String[] info = details.split("//");
+                    String item = info[0].trim();
```
You could have split the string with whitespace included like: details.split(" // ") to avoid trimming the string later

ignatiusoey09 reviewed (commented):
LGTM! Just some minor nits to fix 👍

leroychiu20 committed:
Implement javafx halfway

leroychiu20 committed:
Add GUI to alex

leroychiu20 committed:
Improve GUI

leroychiu20 committed:
Merge branch 'branch-Level-10'

leroychiu20 committed:
Add assertion statements

I as a programmer may introduce bugs into my code.

Adding assertion statements will help me to know when I introduced unwanted bugs into my code while editing the code.

Assertion statements should be used, not catching exceptions, because exceptions are meant to indicate that the user or environment messed up.

leroychiu20 committed:
Improve code quality

To make the code more readable, code quality must be improved. This is to ensure that someone reading the code is able to understand what is going on at all times.

Let's edit any lines of code that has deep nesting to code without, avoid ambiguous terms, and add comments before each method or class to explain its purpose.

leroychiu20 committed:
Merge pull request #1 from leroychiu20/branch-A-Assertions

Add assertion statements

leroychiu20 committed:
Merge branch 'master' of https://github.com/leroychiu20/ip

leroychiu20 committed:
Merge branch 'master' into branch-A-CodeQuality

leroychiu20 committed:
Merge pull request #2 from leroychiu20/branch-A-CodeQuality

Improve code quality

leroychiu20 committed:
Merge branch 'master' of https://github.com/leroychiu20/ip

leroychiu20 committed:
Create github-actions-demo.yml

leroychiu20 committed:
Merge branch 'master' of https://github.com/leroychiu20/ip

leroychiu20 committed:
Add tag feature

leroychiu20 committed:
Incorporate parser class

leroychiu20 committed:
Add welcome message and more parses

leroychiu20 committed:
Edit and add more tests

leroychiu20 committed:
Merge branch 'branch-A-MoreTesting'

leroychiu20 committed:
Add Ui.png

leroychiu20 committed:
Add user guide

leroychiu20 committed:
Create new jar file

leroychiu20 committed:
Change Readme

leroychiu20 committed:
Edit some typing mistakes
